src/main/kotlin/org/jetbrains/research/testspark/helpers/psi/kotlin/KotlinPsiMethodWrapper.kt

```diff
@@ -0,0 +1,108 @@
+package org.jetbrains.research.testspark.helpers.psi.kotlin
+
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.psi.util.parentOfType
+import org.jetbrains.kotlin.idea.refactoring.isInterfaceClass
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtClassOrObject
+import org.jetbrains.kotlin.psi.KtFunction
+import org.jetbrains.kotlin.psi.KtPrimaryConstructor
+import org.jetbrains.kotlin.psi.KtSecondaryConstructor
+import org.jetbrains.kotlin.psi.KtTypeReference
+import org.jetbrains.kotlin.psi.psiUtil.containingClassOrObject
+import org.jetbrains.research.testspark.helpers.psi.PsiClassWrapper
+import org.jetbrains.research.testspark.helpers.psi.PsiMethodWrapper
+
+class KotlinPsiMethodWrapper(val psiFunction: KtFunction) : PsiMethodWrapper {
+
+    override val name: String get() = psiFunction.name ?: ""
+
+    override val text: String? = psiFunction.text
+
+    override val containingClass: PsiClassWrapper? = psiFunction.run {
+        parentOfType<KtClass>()?.let { KotlinPsiClassWrapper(it) }
+    }
+
+    override val containingFile: PsiFile = psiFunction.containingFile
+
+    override val methodDescriptor: String
+        get() = psiFunction.run {
+            val parameterTypes = valueParameters.joinToString("") { generateFieldType(it.typeReference) }
+            val returnType = generateReturnDescriptor(psiFunction)
+            return "$name($parameterTypes)$returnType"
+        }
+
+    override val signature: String
+        get() = psiFunction.run {
+            val bodyStart = bodyExpression?.startOffsetInParent ?: textLength
+            text.substring(0, bodyStart).replace('\n', ' ').trim()
+        }
+
+    val parameterList = psiFunction.valueParameterList
+
+    val isPrimaryConstructor: Boolean = psiFunction is KtPrimaryConstructor
+
+    val isSecondaryConstructor: Boolean = psiFunction is KtSecondaryConstructor
+
+    val isTopLevelFunction: Boolean = psiFunction.containingClassOrObject == null
+
+    val isDefaultMethod: Boolean = psiFunction.run {
+        val containingClass = PsiTreeUtil.getParentOfType(this, KtClassOrObject::class.java)
+        val containingInterface = containingClass?.isInterfaceClass()
+        name != "<init>" && bodyExpression != null && containingInterface == true
+    }
+
+    override fun containsLine(lineNumber: Int): Boolean {
+        val psiFile = psiFunction.containingFile
+        val document = PsiDocumentManager.getInstance(psiFile.project).getDocument(psiFile) ?: return false
+        val textRange = psiFunction.textRange
+        val startLine = document.getLineNumber(textRange.startOffset) + 1
+        val endLine = document.getLineNumber(textRange.endOffset) + 1
```

Comment:
Why do we have a `+1` in both cases? Could you place a comment in the code to explain it so that the explanation persists?